src/applications/edu-benefits/complaint-tool/reducers/schoolSelect.js

```diff
@@ -0,0 +1,110 @@
+import {
+  LOAD_SCHOOLS_STARTED,
+  LOAD_SCHOOLS_SUCCEEDED,
+  // LOAD_SCHOOLS_FAILED,
+  SEARCH_INPUT_CHANGE,
+  SELECT_INSTITUTION
+} from '../actions/schoolSearch';
+
+const initialState = {
+  currentPageNumber: 1,
+  institutions: [],
+  institutionQuery: '',
+  institutionSelected: {},
+  pagesCount: 0,
+  searchInputValue: '',
+  searchResultsCount: 0,
+  showInstitutions: false,
+  showInstitutionsLoading: false,
+  showPagination: false,
+  showPaginationLoading: false
+};
+
+export default function schoolSearch(state = initialState, action) {
+  switch (action.type) {
+    case LOAD_SCHOOLS_STARTED: {
+      const currentPageNumber = action.page ? action.page : 1;
+      const institutionQuery = action.institutionQuery;
+      const institutions = [];
+      const institutionSelected = {};
+      const searchResultsCount = action.page ? state.searchResultsCount : 0;
+      const showInstitutions = false;
+      const showInstitutionsLoading = !action.page;
+      const showPagination = action.page ? action.page > 1 : false;
+      const showPaginationLoading = !!action.page;
+
+      return {
+        ...state,
+        currentPageNumber,
+        institutionQuery,
+        institutions,
+        institutionSelected,
+        searchResultsCount,
+        showInstitutions,
+        showInstitutionsLoading,
+        showPagination,
+        showPaginationLoading
+      };
+    }
+
+    case LOAD_SCHOOLS_SUCCEEDED: {
+      const {
+        data = [],
+        meta
+      } = action.payload;
+
+      const searchResultsCount = meta.count;
+
+      const institutionQuery = action.institutionQuery;
+      const institutions = data.map(({ attributes }) => {
+        const { city, country, facilityCode, name, state: institutionState, zip } = attributes;
+        return { city, country, facilityCode, name, state: institutionState, zip };
```

Comment:
It looks like this is used in 2 places, would it be of interest to pull this logic into a helper and reuse here and in `transformInstitutionsForSchoolSelectField`?

Reply:
Usually I don't dry out code unless it's a large block that's repeated more than 3 or 4 times. In this case I'm using destructuring to document the object structure which I think makes the code easier to follow and edit later.